test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+
+  forall i in 2..<n do
+    F[i] = (E[i-1] + E[i+1]) / 2;
+
+  writeln("After the safe averaging loop, F is: ", F);
+
+/* 
+
+  Part of the reason the Chapel compiler does not prevent writing
+  parallel loops with races is that it can be difficult to determine
+  whether a given loop is safe to parallelize or not.  For example,
+  the following variation on the original loop would be safe since it
+  only writes to even elements and reads from odd ones:
+
+*/
+
+  D = [i in 1..n] (i**2): real;
+
+  writeln("Before the third averaging loop, D is: ", D);
+  forall i in 2..<n by 2 do
+    D[i] = (D[i-1] + D[i+1]) / 2;
+
+  writeln("After the third averaging loop, D is: ", D);
+
+/*
+
+  Distinguishing between loops that are parallel-safe versus not is
+  generally intractable, so rather than attempting to make that
+  judgment, Chapel trusts the programmer to use the loop form they
+  want.  Moreover, for some parallel computations, race conditions can
+  be benign or acceptable, so Chapel does not want to prevent a user
+  from writing such computations.
+
+  An important thing to remember about Chapel's forall-loops is that
+  they are essentially invocations of parallel iterators.  In
+  practice, those parallel iterators are themselves often written in
+  terms of the data-parallel loop forms above, or the more explicit
+  task-parallel coforall-loops that we'll cover next.
+
+
+  Task-Parallel Loops
+  ===================
+
+  Chapel has a single task-parallel loop form, the ``coforall`` loop:
+  
+  coforall loops
+  --------------
+
+  In most respects, the coforall-loop is the simplest parallel loop
+  form to explain in Chapel.  It literally creates a distinct Chapel
+  task for each iteration of the loop and then waits until those tasks
+  have each completed executing their copy of the loop body before
+  proceeding.  Mnemonically, ``coforall`` can be thought of as a
+  "concurrent forall-loop".  For example, the following coforall loop
+  will create four tasks, one for each iteration of the loop:
+
+*/
+
+  config const numTasks = 4;
+  var total: atomic int;
+  
+  coforall tid in 1..numTasks do
+    total.add(tid);
+
+  writeln("The total of the integers 1..", numTasks, " is ", total.read());
+
+/*
+
+  Because each iteration is executed by a separate task,
+  coforall-loops can synchronize between distinct iterations of the
+  loop, unlike the data-parallel loop forms.  This makes
+  coforall-loops useful as a means of creating an arbitrary number of
+  tasks that are independently doing the same, or similar, things.  In
+  practice, the parallel iterators used to define ``forall`` loops are
+  often implemented in terms of coforall-loops that create a task per
+  processor core and/or locale.  As an example, the following
+  coforall-loop creates a task per processor core across all locales:
+
+*/
+
+
+  coforall loc in Locales {
+    on loc {
+      coforall tid in 1..numTasks {
```

Comment:
`Locales` is the Chapel language array of locales, but here `numTasks` doesn't represent the number of cores as your comment implies, but it is 4, defined above, right?

If it were switched to `here.maxTaskPar`, that would be more accurate to the comment, but then gets a little uglier... 

If it were me, I'd probably put `here.maxTaskPar` or maybe set `numTasks` to `here.maxTaskPar` and explain that line in a comment?